app/domain/authentication/authn_jwt/fetch_jwt_signing_key_factory.rb

```diff
@@ -0,0 +1,44 @@
+module Authentication
+  module AuthnJwt
+    # Factory that returns the interface implementation of FetchSigningKey
+    class FetchJwtSigningKeyFactory
+
+      def initialize(authenticator_parameters)
+        @authenticator_input = authenticator_parameters
+        @fetch_provider_uri_signing_key = Authentication::AuthnJwt::FetchProviderUriSigningKey.new(@authenticator_input)
+        @fetch_jwks_uri_signing_key = Authentication::AuthnJwt::FetchJwksUriSigningKey.new(@authenticator_input)
+      end
+
+      def create
+        validate_key_configuration
+
+        if provider_uri_is_valid?
+          @fetch_provider_uri_signing_key
+        elsif jwks_uri_is_valid?
+          @fetch_jwks_uri_signing_key
+        end
+      end
+
+      private
+
+      def validate_key_configuration
+        if (provider_uri_is_valid? and jwks_uri_is_valid?) or
+          (!provider_uri_is_valid? and !jwks_uri_is_valid?)
+          raise Errors::Authentication::AuthnJwt::InvalidUriConfiguration.new(
+            PROVIDER_URI_RESOURCE_NAME,
+            JWKS_URI_RESOURCE_NAME
+          )
+        end
+      end
+
+      def provider_uri_is_valid?
+        @provider_uri_resource_exists ||= @fetch_provider_uri_signing_key.has_valid_configuration?
+      end
```

Comment:
memozation methods should have the same name as the objects they memoize.

Reply:
My bad, forgot to rename that too

Reply:
note that you have here 3 boolean names:
1. `provider_uri_is_valid?`
2. `provider_uri_resource_exists`
3. `provider_uri_has_valid_configuration?

they should all be the same as far as i can tell